pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -713,6 +722,30 @@ func convertToPrincipal(subject *rbacproto.Subject, forTCPFilter bool) *policypr
 		subject.Properties[attrRequestClaimGroups] = subject.Group
 	}
 
+	namesBinding := convertBindingField(subject.Names, attrSrcPrincipal, forTCPFilter)
```

Comment:
Do you need to check if it returns nil?

Reply:
appendID() and appendNotID() have nil checks. I think that should be enough.

Reply:
Good point. Done. Thanks!